src/ibm/hypervisor-network-mgr-src/hyp_ethernet_interface.cpp

```diff
@@ -21,6 +21,191 @@ constexpr char biosIntType[] =
 constexpr char biosEnumType[] =
     "xyz.openbmc_project.BIOSConfig.Manager.AttributeType.Enumeration";
 
+biosTableRetAttrValueType
+    HypEthInterface::getAttrFromBiosTable(const std::string& attrName)
+{
+    constexpr auto BIOS_SERVICE = "xyz.openbmc_project.BIOSConfigManager";
+    constexpr auto BIOS_OBJPATH = "/xyz/openbmc_project/bios_config/manager";
+    constexpr auto BIOS_MGR_INTF = "xyz.openbmc_project.BIOSConfig.Manager";
+
+    try
+    {
+        using getAttrRetType =
+            std::tuple<std::string, std::variant<std::string, int64_t>,
+                       std::variant<std::string, int64_t>>;
+        getAttrRetType ip;
+        auto method = bus.new_method_call(BIOS_SERVICE, BIOS_OBJPATH,
+                                          BIOS_MGR_INTF, "GetAttribute");
+
+        method.append(attrName);
+
+        auto reply = bus.call(method);
+
+        std::string type;
+        std::variant<std::string, int64_t> currValue;
+        std::variant<std::string, int64_t> defValue;
+        reply.read(type, currValue, defValue);
+        return currValue;
+    }
+    catch (const sdbusplus::exception::SdBusError& ex)
+    {
+        log<level::ERR>("Failed to get the attribute value from bios table",
+                        entry("ERR=%s", ex.what()));
+    }
+    return "";
+}
+
+void HypEthInterface::watchBaseBiosTable()
+{
+    auto BIOSAttrUpdate = [this](sdbusplus::message::message& m) {
+        std::map<std::string, std::variant<BiosBaseTableType>>
+            interfacesProperties;
+
+        std::string objName;
+        m.read(objName, interfacesProperties);
+
+        // Check if the property change signal is for BaseBIOSTable property
+        // If found, proceed; else, continue to listen
+        if (!interfacesProperties.contains("BaseBIOSTable"))
+        {
+            // Return & continue to listen
+            return;
+        }
+        log<level::INFO>("BaseBIOSTable - property changed");
+
+        // Check if the IP address has changed (i.e., if current ip address in
+        // the biosTableAttrs data member and ip address in bios table are
+        // different)
+
+        // the no. of interface supported is two
+        constexpr auto MAX_INTF_SUPPORTED = 2;
```

Comment:
Is this not 0 and 1 ? So max intf supported will be 1

Reply:
No sunitha..
Max  number of interfaces is two.. Just the indexing is 0 and 1..
So I think this MAX_INTF_SUPPORTED should be 2 only